src/main.rs

```diff
@@ -157,14 +168,15 @@ fn main() {
                     }
                     bot::Task::FetchComments(user, change_id, message) => {
                         let gerrit_config = config.gerrit.clone();
-                        let gerrit_hostname = config.gerrit.hostname.clone();
-                        let gerrit_change = match gerrit::query(
-                            &gerrit_config.hostname,
-                            gerrit_config.port,
-                            &gerrit_config.username,
-                            &gerrit_config.priv_key_path,
-                            &change_id,
-                        ) {
+                        let gerrit_host = config.gerrit.host.clone();
```

Comment:
Hmm. I would prefer to move this logic to a different thread, and communicate through a pipe.

Reply:
I can give that a shot.

Reply:
Have a look here, please:  https://github.com/fbenkstein/gerritbot-rs/commit/f7fd0d3a278b569786b0004935b93a050bf54c82

Reply:
I like the approach, you should push directly to this PR!

Reply:
Thanks @fbenkstein. I used your idea but without using unsafe. I just introduced another function: reconnect.